tests/test.sh

```diff
@@ -1,5 +1,17 @@
 #!/bin/bash
 
+# Ensure this can be called from the root of the repo as `tests/test.sh`
+# It cannot be called from other directories, and if it is, errors out immediately
+CALLINGDIR=`pwd`
+BASENAME=`basename $CALLINGDIR`
+if [ $BASENAME != 'tests' ]; then
+  cd tests
```

Comment:
Instead of hardcoding all of this, you could use something like `cd $(dirname "$0")` which should then handle execution from any dir.